classes/class-dintero-checkout-callback.php

```diff
@@ -0,0 +1,194 @@
+<?php
+/**
+ * Handle callbacks from Dintero.
+ *
+ * @package Dintero_Checkout/Classes
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class for handling callbacks from Dintero.
+ */
+class Dintero_Checkout_Callback {
+
+	/**
+	 * Register callback action.
+	 */
+	public function __construct() {
+		add_action( 'woocommerce_api_dintero_callback', array( $this, 'callback' ) );
+	}
+
+	/**
+	 * Handle the callback from Dintero.
+	 *
+	 * @return void
+	 */
+	public function callback() {
+		$merchant_reference = filter_input( INPUT_GET, 'merchant_reference', FILTER_SANITIZE_STRING ); /* The merchant_reference is guaranteed to always be available. */
+		$transaction_id     = filter_input( INPUT_GET, 'transaction_id', FILTER_SANITIZE_STRING ); /* The transaction_id is guaranteed unless 'error' is set. */
+		$error              = filter_input( INPUT_GET, 'error', FILTER_SANITIZE_STRING );
+
+		/* If the 'order_key' does not exist, we cannot identify the WC order. */
+		$order_key = filter_input( INPUT_GET, 'key', FILTER_SANITIZE_STRING );
+		if ( empty( $order_key ) ) {
+			Dintero_Logger::log( sprintf( 'CALLBACK ERROR [order_key]: No order key was found (transaction ID: %s). Cannot identify the WC order. ', ( $transaction_id ) ? $transaction_id : 'Not available' ) );
+			return;
+		}
+
+		$order_id = wc_get_order_id_by_order_key( $order_key );
+		if ( empty( $order_id ) ) {
```

Comment:
Could be advantageous to use the new type system, e.g. `int $order_id` instead of in the docblocs.

Reply:
I think this is a great idea if the minimum required PHP version is 7.4. I'll discuss this with my colleagues.

Reply:
Ok. I'm not sure how the ecosystem is. I see 7.4 is soon deprecated, but I guess a lot of people still use it?

Reply:
Practically, the minimum required PHP version right now is 7.0.

https://woocommerce.com/document/update-php-wordpress/

Reply:
Oh, ok, I guess it's best to not break pre-7.4 compatibility on purpose, then :+1:

Reply:
I'll update the required PHP version in the readme.txt file to PHP 7 once I've tested it with PHP 7 and 7.4.